### ISSUE TITLE
References to NamespaceSystem directly contravene coredns supplicant chart

### ISSUE THREAD

**Issue by ElDiabloRojo** [containous/maesh#357]
@dtomcej  please can you provide some insight on the following:

When deploying coredns via the maesh chart the namespace for all specs [here](https://github.com/containous/maesh/tree/master/helm/chart/maesh/templates/dns/coredns) reference ```{{ .Release.Namespace }}``` which is usually going to be "maesh". 
This seems to contradict the static configuration in the metav1 library see ```metav1.NamespaceSystem``` in [here](https://github.com/containous/maesh/blob/master/internal/k8s/client.go).

Please let me know if what I'm describing here isn't clear, you seem to be the main contributor to both of these areas in the project so perhaps you can clarify the intended deployment style as it is not documented anywhere. 

**TLDR**; the coredns chart is deployed to the release namespace and controller application looks for coredns in the kube-system namespace.

here is the log output from the failing ```maesh-prepare``` init container:

```
time="2019-11-14T10:18:53Z" level=info msg="Configuration loaded from flags."
time="2019-11-14T10:18:53Z" level=debug msg="Starting maesh prepare..."
time="2019-11-14T10:18:53Z" level=debug msg="Using masterURL: \"\""
time="2019-11-14T10:18:53Z" level=debug msg="Using kubeconfig: \"\""
time="2019-11-14T10:18:53Z" level=debug msg="Building Kubernetes Client..."
time="2019-11-14T10:18:53Z" level=debug msg="Building SMI Access Client..."
time="2019-11-14T10:18:53Z" level=debug msg="Building SMI Specs Client..."
time="2019-11-14T10:18:53Z" level=debug msg="Building SMI Split Client..."
time="2019-11-14T10:18:53Z" level=info msg="Checking Cluster..."
time="2019-11-14T10:18:53Z" level=info msg="Checking CoreDNS..."
time="2019-11-14T10:18:53Z" level=debug msg="Get CoreDNS version..."
time="2019-11-14T10:18:53Z" level=debug msg="coredns does not exist in namespace kube-system"
time="2019-11-14T10:18:53Z" level=info msg="Checking KubeDNS..."
time="2019-11-14T10:18:53Z" level=debug msg="Get KubeDNS version..."
time="2019-11-14T10:18:53Z" level=debug msg="kube-dns does not exist in namespace kube-system"
2019/11/14 10:18:53 command prepare error: error during cluster check: no core dns service available for installing maesh: <nil>
```

**Comment by dtomcej** [comment]
Hello @ElDiabloRojo,

The Maesh prep tool checks for 2 options:

1. Your cluster uses coreDNS by default, and it will be present in kube-system. This is the ideal situation, as Maesh is directly able to patch and use this.
2. Your cluster uses KubeDNS by default, and it will be present in kube-system. In this situation, we install an instance of coreDNS into the Maesh namespace, and use a shim to redirect DNS requests from kubeDNS to coreDNS.

Either way, you need to have coreDNS or kubeDNS active and installed in your kube-system namespace as required by our prerequisites: https://docs.mae.sh/#prerequisites

The error message you are seeing is stating that it cannot find either kubeDNS or coreDNS in your cluster.

**Comment by dtomcej** [comment]
@ElDiabloRojo did this answer your question?

Please let us know and we will close this issue.

**Comment by dtomcej** [comment]
Hello @ElDiabloRojo, we will consider this issue closed.

Please feel free to open a new issue if you encounter one!
